src/web/demo.rkt

```diff
@@ -395,8 +429,8 @@
       (define formula-str (hash-ref post-data 'formula))
       (define formula (read-syntax 'web (open-input-string formula-str)))
       (define seed* (hash-ref post-data 'seed))
-      (define job-id (sha1 (open-input-string (~s 'sample formula seed*))))
-      (semaphore-wait (run-sample job-id formula seed*))
+      (define job-id (compute-job-id (list 'sample formula seed*)))
+      (run-work #t job-id (list 'sample formula seed*))
       (define result (hash-ref *completed-jobs* job-id))
       (define pctx (job-result-backend result))
```

Comment:
I feel like this bit, where you get the job result from the `*completed-jobs*`, could also happen inside `run-work`, at least if you run it in sync mode.

Reply:
I think you are right. Though can I circle back to this after these 3 PRs as I think I change this in later PRs.